addons/sms_twilio/models/sms_sms.py

```diff
@@ -0,0 +1,27 @@
+from odoo import fields, models
+
+from odoo.addons.sms.models.sms_sms import SmsSms as SmsSmsOriginal
+
+
+class SmsSms(models.Model):
+    _inherit = 'sms.sms'
+
+    IAP_TO_SMS_FAILURE_TYPE = SmsSmsOriginal.IAP_TO_SMS_FAILURE_TYPE | {
+        'twilio_authentication': 'sms_twilio_authentication',
+    }
```

Comment:
Note
Probably too annoying of a change for stable as we need to keep the globals anyway, but could be nice to move all these api-specific mappings to the "API" classes.